From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Thu, 11 Jun 2026 17:54:13 +0530
Subject: Fix drm_exec_init signature mismatch between 5.15 and 6.6

The drm_exec_init function has different signatures across kernel
versions: 5.15 backport uses 3 arguments, while 6.6 native kernel
uses 2 arguments. This caused "too few arguments" compile errors.

Make the prototype conditional based on which kernel version is
being built, ensuring call sites match the actual function signature.

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 include/drm/drm_exec.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/drm/drm_exec.h b/include/drm/drm_exec.h
index aa786b8..b09e9f0 100644
--- a/include/drm/drm_exec.h
+++ b/include/drm/drm_exec.h
@@ -135,7 +135,11 @@ static inline bool drm_exec_is_contended(struct drm_exec *exec)
 	return !!exec->contended;
 }
 
+#if defined(BPM_DRM_EXEC_INIT_ARG3_NOT_PRESENT) && !defined(CPTCFG_BUILD_XE_DRM_EXEC)
+void drm_exec_init(struct drm_exec *exec, u32 flags);
+#else
 void drm_exec_init(struct drm_exec *exec, u32 flags, unsigned nr);
+#endif
 void drm_exec_fini(struct drm_exec *exec);
 bool drm_exec_cleanup(struct drm_exec *exec);
 int drm_exec_lock_obj(struct drm_exec *exec, struct drm_gem_object *obj);
-- 
2.43.0
